docs/workflow.md

```diff
@@ -26,6 +26,18 @@ See the document on saved launch configurations for how these are processed
 when the job is launched, and the API validation involved in building
 the launch configurations on workflow nodes.
 
+#### Workflows as Workflow Nodes
+
+A workflow can be added as a node in another workflow. The child workflow is the associated
+`unified_job_template` that the node references, when that node is added to the parent workflow.
+When the parent workflow dispatches that node, then the child workflow will begin running, and
+the parent will resume execution of that branch when the child workflow finishes.
+Branching into success / failed pathways is decided based on the status of the child workflow.
```

Comment:
@AlanCoding (cc @Spredzy) - When I originally tested workflows, I remember that it was pretty much impossible to get them into a failure state (I never could). Has the definition of workflow success / failure / error changed since workflows were originally written? Wondering if it's still basically impossible for workflows to fail.

Reply:
Yes, some time ago jangsutsr implemented workflow failure condition. See https://github.com/ansible/awx/pull/382

The history of workflow status was this:
 - pre-release in initial implementation we had the "completed" state, so they would either be completed or error
 - various ideas were floated around about user-defined failure, but nothing was agreed on, except that we probably wanted this notion in the future, and a result of this was that the (blue) completed state was actually walked back, the implementation because either "success" or "error"
 - some time after that, the failure condition was implemented, so it became "success", "failure", or "error"

You could also add "cancel" to these. I have some pending patches for bugs with the cancel status too.

Because of the current definition of workflow failure, the failure scenario for sliced job just _happens_ to already be identical to if they were not sliced. All nodes are leaf nodes, so failure of any fails the workflow, just as any failure on a host fails the playbook.

In other words, the branching of the network with workflows-in-workflows required no implementation, and is taken to already work with no change to existing logic. Sure, test it anyway, if you see fit.